### PR TITLE
Remove @Deprecated JvmPluginsHelper.configureDocumentationVariantWithArtifact method

### LINKED ISSUE #21679: Remove @Deprecated JvmPluginsHelper.configureDocumentationVariantWithArtifact method

### PR DESCRIPTION
**Pull request by tresat**
Fixes #21679

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java [hunk 1]
```diff
@@ -171,43 +170,6 @@ public static void configureJavaDocTask(@Nullable String featureName, SourceSet
         }
     }
 
-    /**
-     * @deprecated Use {@link #configureDocumentationVariantWithArtifact(String, String, String, List, String, Object, AdhocComponentWithVariants, ConfigurationContainer, TaskContainer, ObjectFactory, FileResolver)}
-     * instead. Passing {@code null} for the FileResolver will not be legal after this is removed, please provide one.
-     */
-    @Deprecated
-    public static void configureDocumentationVariantWithArtifact(
```
Comment (jvandort) [inline subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java:179]: There is another method with the same name in `DefaultJvmVariantBuilder`, and it looks quite similar. Do you know if these can be merged?

**Review by jvandort — APPROVED** [review]

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java (hunk 1 above)
Comment (tresat) [reply subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java:179]: These do look like they would produce identical variants, at least at first glance after following what attributes they would create.  I'm going to explore merging them in a separate PR.  Good catch!

**Pull request merged**